sql/core/src/main/scala/org/apache/spark/sql/execution/exchange/BroadcastExchangeExec.scala

```diff
@@ -74,7 +74,9 @@ case class BroadcastExchangeExec(
     child: SparkPlan) extends BroadcastExchangeLike {
   import BroadcastExchangeExec._
 
-  override val runId: UUID = UUID.randomUUID
+  private val groupId = sparkContext.getLocalProperty(SparkContext.SPARK_JOB_GROUP_ID)
```

Comment:
is it reliable? does the local property always contain the group id of the query that this broadcast belongs to?

Reply:
For STS, I think so. Building this physical plan preformed in a thread of `HiveServer2-Background-Pool`, the same thread of setting up the job group Id, parsing, analyzing, and optimizing, etc. Still keeping `UUID.randomUUID` is to avoid UT failures IIUC.